packages/video_player_avplayer/tizen/src/plus_player.cc

```diff
@@ -0,0 +1,666 @@
+  plusplayer::PlayerMemento memento;
+  if (!player_->GetMemento(&memento)) {
+    LOG_ERROR("[PlusPlayer] Fail to get player memento.");
+  }
+
+  if (memento.is_live) {
+    std::string str = player_->GetStreamingProperty("GET_LIVE_DURATION");
+    if (str.empty()) {
+      LOG_ERROR("[PlusPlayer] Fail to get live duration.");
+    }
+    std::vector<std::string> time_str = split(str, '|');
+    int64_t start_time = std::stoll(time_str[0].c_str());
+    int64_t end_time = std::stoll(time_str[1].c_str());
+
+    if (position < start_time || position > end_time) {
+      on_seek_completed_ = nullptr;
+      LOG_ERROR("[PlusPlayer] position out of range.");
+      return;
+    }
+
+    if (!player_->Seek(position)) {
+      on_seek_completed_ = nullptr;
+      LOG_ERROR("[PlusPlayer] Fail to seek.");
+    }
+  } else {
+    if (!player_->Seek(position)) {
+      on_seek_completed_ = nullptr;
+      LOG_ERROR("[PlusPlayer] Fail to seek.");
+    }
+  }
+}
+
+int64_t PlusPlayer::GetPosition() {
+  uint64_t position = 0;
+  plusplayer::State state = player_->GetState();
+  if (state == plusplayer::State::kPlaying ||
+      state == plusplayer::State::kPaused) {
+    if (!player_->GetPlayingTime(&position)) {
+      LOG_ERROR("[PlusPlayer] Fail to get the current playing time.");
+    }
+  }
+  return static_cast<int64_t>(position);
+}
+
+int64_t PlusPlayer::GetDuration() {
+  int64_t duration = 0;
+  if (player_->GetState() >= plusplayer::State::kTrackSourceReady) {
+    plusplayer::PlayerMemento memento;
+    if (!player_->GetMemento(&memento)) {
+      LOG_ERROR("[PlusPlayer] Fail to get player memento.");
+    }
+
+    if (memento.is_live) {
+      std::string str = player_->GetStreamingProperty("GET_LIVE_DURATION");
+      if (str.empty()) {
+        LOG_ERROR("[PlusPlayer] Fail to get live duration.");
+        return duration;
+      }
+      std::vector<std::string> time_str = split(str, '|');
+      int64_t start_time = std::stoll(time_str[0].c_str());
+      int64_t end_time = std::stoll(time_str[1].c_str());
+
+      duration = end_time - start_time;
+    } else {
+      if (!player_->GetDuration(&duration)) {
+        LOG_ERROR("[PlusPlayer] Fail to get the duration.");
+      }
+    }
+  }
+
+  LOG_INFO("[PlusPlayer] Video duration: %lld.", duration);
+  return duration;
+}
+
+void PlusPlayer::GetVideoSize(int32_t *width, int32_t *height) {
+  if (player_->GetState() >= plusplayer::State::kTrackSourceReady) {
+    bool found = false;
+    std::vector<plusplayer::Track> tracks = player_->GetActiveTrackInfo();
+    for (auto track : tracks) {
+      if (track.type == plusplayer::TrackType::kTrackTypeVideo) {
+        *width = track.width;
+        *height = track.height;
+        found = true;
+      }
+    }
+    if (!found) {
+      LOG_ERROR("[PlusPlayer] Failed to get video size.");
+    } else {
+      LOG_INFO("[PlusPlayer] Video widht: %d, height: %d.", *width, *height);
+    }
+  }
+}
+
+bool PlusPlayer::isReady() {
```

Comment:
```suggestion
bool PlusPlayer::IsReady() {
```

Also `GetTrackInfo`.

Reply:
Fixed.